oci/spec_opts_windows.go

```diff
@@ -100,3 +101,26 @@ func escapeAndCombineArgs(args []string) string {
 	}
 	return strings.Join(escaped, " ")
 }
+
+func WithDevices(devicePath, containerPath, permissions string) SpecOpts {
+	return func(_ context.Context, _ Client, _ *containers.Container, s *Spec) error {
+		if containerPath != "" {
+			return fmt.Errorf("unexpected containerPath %s", containerPath)
+		}
+
+		if permissions != "" {
+			return fmt.Errorf("unexpected permissions %s", permissions)
+		}
+
+		deviceParts := strings.Split(devicePath, "/")
```

Comment:
This is the simple approach, matching what Docker does, per https://github.com/kubernetes/kubernetes/issues/97739#issuecomment-757509573.

Docker only accepts `class`, rejecting other options. Conversely, now I look at it, they allow `id` to contain `/`... I'll need to have a look at the consuming code in hcsshim and find some examples, as I can see that might be necessary for device-instance assignment.

Reply:
I tracked down `://`, and ended up rewriting this backed on the learnings.